lib/datadog/tracing/contrib/roda/instrumentation.rb

```diff
@@ -0,0 +1,84 @@
+require_relative '../../metadata/ext'
+require_relative 'ext'
+require_relative '../analytics'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module Roda
+        # Instrumentation for Roda
+        module Instrumentation
+
+          def _roda_handle_main_route
+            instrument(Ext::SPAN_REQUEST) { super }
+          end
+
+          def call
+            instrument(Ext::SPAN_REQUEST) { super }
```

Comment:
Can move the [error handling logic](https://github.com/DataDog/dd-trace-rb/pull/2368/files#diff-882be2b4ec172fe41860d166e42fd865e3b6610a7ee43d0e4b382a03033ea0cbR50-R57) to the caller

```
          def call
            instrument(Ext::SPAN_REQUEST) do |span|
              super
            rescue => e
              # ... Handle error
            end
          end
```
So you could be confident yielding in the block without nested begin/rescue.
```
response = yield span
```

Reply:
It looks like in the Roda code, dispatch can land on one of these methods ([seen in this conditional](https://github.com/jeremyevans/roda/blob/04f7e70215b1ec1151286095b8825626335d5604/lib/roda.rb#L456))
- [call](https://github.com/jeremyevans/roda/blob/04f7e70215b1ec1151286095b8825626335d5604/lib/roda.rb#L522)
- [_roda_handle_main_route ](https://github.com/jeremyevans/roda/blob/04f7e70215b1ec1151286095b8825626335d5604/lib/roda.rb#L493)

Would I need to duplicate the rescue block and handle the error in both since either one of these can handle the dispatch?

Reply:
As an update to this, I wasn't able to move the error handling logic anywhere else because it interfered with the errors produced by the underlying app.